### DISCUSSION TITLE
Memory leak issue on production.

### DISCUSSION THREAD

**Discussion by prasann004** [socketio/socket.io#4821] [Q&A]
This is my server code. I have been facing memory leaks in production.
"socket.io": "2.2.0",
"socket.io-redis": "^5.2.0". 
I am relatively new to socket.io. Any suggestions get rid of this memory leak.

<img width="879" alt="SocketServer" src="https://github.com/socketio/socket.io/assets/47444136/e9bc6ed5-dbcf-470f-b1ba-5283dcaba7df">
<img width="1297" alt="CPU MemoryUtilisation" src="https://github.com/socketio/socket.io/assets/47444136/d7eb58e8-6848-4b3f-825b-47353a7f8baf">

**Comment by darrachequesne** [comment]
Hi! Unfortunately, this is not sufficient for us to be able to help you.

Could you please provide a complete server code? Do you have a heap snapshot?
